samples/livr-emitter/src/schema.ts

```diff
@@ -0,0 +1,290 @@
+/**
+ * Base interface for all LIVR rules.
+ * Extend and discriminate with `rule` property.
+ */
+interface LIVRRuleBase {
+  rule: string;
+  error_code?: string;
+}
+
+// --- Common rules ---
+export interface RequiredRule extends LIVRRuleBase {
+  rule: "required";
+  error_code?: "REQUIRED";
```

Comment:
Should we create an enum to represent the error codes? Many of them are used in multiple places.